File: DotNet/VS2003/Windows/Other/Microsoft Enterprise Instrumentation Framework/Event Sources/cs/Event Source API.cs

// Copyright (c) Microsoft Corporation.  All rights reserved.

using System;
using System.ComponentModel;
using Microsoft.EnterpriseInstrumentation;
using Microsoft.EnterpriseInstrumentation.Schema;
using TraceToolEventSink ;

namespace EventSourceAPI
{
	[RunInstaller(true)]
	public class MyProjectInstaller : ProjectInstaller {}

	/// <summary>
	/// ExplicitTrace class illustrates 3 examples of explicit tracing
	/// </summary>
	class EventSourceAPI
	{
		private static EventSource myEventSource = new EventSource("EventSourceAPI");
		/// <summary>
		/// The main entry point for the application.
		/// </summary>
		[STAThread]
		static void Main(string[] args)
		{

         //TraceToolEventSink.EIFEventSink, TraceToolEventSink, Version=1.0.1622.23936, Culture=neutral, PublicKeyToken=35bb89f90be2164b
         System.Console.WriteLine (typeof(TraceToolEventSink.EIFEventSink).AssemblyQualifiedName);


			// explicit
			TraceMessageEvent e1 = new TraceMessageEvent();
			e1.Message = "Costly Message";
			myEventSource.Raise(e1);

			// slightly less costly, if the EventSource will not be enabled much of the time.
			if (myEventSource.IsEnabledForType(typeof(TraceMessageEvent)))
			{
				TraceMessageEvent e2 = new TraceMessageEvent();
				e2.Message = "Less Costly Message";
				myEventSource.Raise(e2);
			}

			// static one liner (which wraps the above code sequence)
			TraceMessageEvent.Raise(myEventSource, "Static One Liner Message");

			// static one liner which is raised through the EventSource.Application EventSource.
			TraceMessageEvent.Raise("Static One Liner Message through the Application event source");

		}
	}
}

File: DotNet/VS2003/Windows/Other/Microsoft Enterprise Instrumentation Framework/Event Sources/vb/Event Source API.vb

' Copyright (c) Microsoft Corporation.  All rights reserved.

Imports System
Imports System.ComponentModel
Imports Microsoft.EnterpriseInstrumentation
Imports Microsoft.EnterpriseInstrumentation.Schema

Namespace EventSourceAPI

    <RunInstaller(True)> Public Class MyProjectInstaller
        Inherits ProjectInstaller
    End Class

    Public Class ConsoleVb
        Private Shared myEventSource As EventSource = New EventSource("EventSourceAPI")
        Public Shared Sub Main()
            ' explicit
            Dim e1 As TraceMessageEvent = New TraceMessageEvent()
            e1.Message = "Costly Message"
            myEventSource.Raise(e1)

            ' slightly less costly, if the EventSource will not be enabled much of the time.
            If myEventSource.IsEnabledForType(GetType(TraceMessageEvent)) Then
                Dim e2 As TraceMessageEvent = New TraceMessageEvent()
                e2.Message = "Less Costly Message"
                myEventSource.Raise(e2)
            End If

            ' static one liner (which wraps the above code sequence)
            TraceMessageEvent.Raise(myEventSource, "Static One Liner Message")

            ' static one liner which is raised through the EventSource.Application EventSource.
            TraceMessageEvent.Raise("Static One Liner Message through the Application event source")

        End Sub
    End Class
End Namespace
